operators/pkg/controller/license/validation/validations.go

```diff
@@ -5,32 +5,26 @@
 package validation
 
 import (
-	estype "github.com/elastic/cloud-on-k8s/operators/pkg/apis/elasticsearch/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/common"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/common/license"
 	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/common/validation"
+	corev1 "k8s.io/api/core/v1"
 	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 )
 
 var log = logf.Log.WithName("license-validation")
 
 var Validations = []Validation{
 	eulaAccepted,
-	requiredFields,
 }
 
 func eulaAccepted(ctx Context) validation.Result {
-	if !ctx.Proposed.Spec.Eula.Accepted {
-		return validation.Result{Allowed: false, Reason: "Please set the field eula.accepted to true to accept the EULA"}
-	}
-	return validation.OK
-}
-
-func requiredFields(ctx Context) validation.Result {
-	if ctx.Proposed.IsTrial() {
+	if ctx.Proposed.Labels[common.TypeLabelName] != license.Type {
 		return validation.OK
 	}
-	err := ctx.Proposed.IsMissingFields()
-	if err != nil {
-		return validation.Result{Allowed: false, Reason: err.Error()}
+
+	if ctx.Proposed.Annotations[license.EULAAnnotation] != license.EULAAcceptedValue {
+		return validation.Result{Allowed: false, Reason: "Please set the annotation elastic.co/eula to accepted to accept the EULA"}
```

Comment:
Maybe quote "accepted"

Reply:
+1 to add quotes.
Maybe it also deserves its constant since it's used three times.